src/colab_superarchives/views.py

```diff
@@ -418,43 +419,86 @@ def get_object(self, *args, **kwargs):
 
         return obj
 
+    def get_context_data(self, **kwargs):
+        user = self.get_object()
+
+        context = self.get_mailling_lists(user)
+
+        context.update(kwargs)
+
+        return super(ManageUserSubscriptionsView,
+                     self).get_context_data(**context)
+
     def post(self, request, *args, **kwargs):
         user = self.get_object()
-        for email in user.emails.values_list('address', flat=True):
-            lists = self.request.POST.getlist(email)
-            info_messages = mailman.update_subscription(email, lists)
-            for msg_type, message in info_messages:
-                show_message = getattr(messages, msg_type)
-                show_message(request, _(message))
 
-        return redirect('user_profile', username=user.username)
+        # If it is an ajax request, get the user mailling lists
+        # to display filtering it by name
+        if request.is_ajax():
+            response = self.search_lists_by_name(user,
+                                                 request.POST.get('listname'))
+        else:
+            for email in user.emails.values_list('address', flat=True):
+                lists = self.request.POST.getlist(email)
+                info_messages = mailman.update_subscription(email, lists)
+                for msg_type, message in info_messages:
+                    show_message = getattr(messages, msg_type)
+                    show_message(request, _(message))
 
-    def get_context_data(self, **kwargs):
+            response = redirect('user_profile', username=user.username)
+
+        return response
+
+    def search_lists_by_name(self, user, listname):
+        context = self.get_mailling_lists(user, listname)
+
+        html = render_to_string(u'accounts/subscription_lists.html',
+                                {'membership': context['membership']})
```

Comment:
@italopaiva , I had to add 'per_page' so the pagination would work with search by list name: https://github.com/colab/colab-superarchives-plugin/blob/master/src/colab_superarchives/views.py#L462